src/ai/backend/manager/actions/types.py

```diff
@@ -24,3 +30,43 @@ class AbstractProcessorPackage(ABC):
     def supported_actions(self) -> list[ActionSpec]:
         """Get the list of action specs that this processors can handle."""
         raise NotImplementedError
+
+
+class MultiEntityFailureHandlingPolicy(enum.StrEnum):
+    ALL_OR_NONE = "all_or_none"
+    PARTIAL_SUCCESS = "partial_success"
+
+
+@dataclass
+class ActionTriggerMeta:
+    action_id: uuid.UUID
+    started_at: datetime
+
+
+@dataclass
+class ActionResultTargetMeta:
+    entity_type: str
+    entity_ids: Optional[list[str]] = None
+    scope: Optional[ScopeId] = None
+
+    @property
+    def entity_id(self) -> Optional[str]:
+        """Return the first entity ID if available, otherwise None."""
+        return self.entity_ids[0] if self.entity_ids else None
```

Comment:
I think everything can be checked in action, so is dataclass really necessary?

Reply:
I want to group related data